### PR TITLE
Use `sentry-cli sourcemaps upload` instead of legacy command

### LINKED ISSUE #10270: Replace `releases files upload-sourcemaps` with `sourcemaps upload` (<34x)
There are at least six, but possibly up to 34 occurrences we need to replace

### PR DESCRIPTION
**Pull request by szokeasaurusrex**
## DESCRIBE YOUR PR

This change removes all references to the legacy `sentry-cli releases files upload-sourcemaps` command, and replaces then with references to `sentry-cli sourcemaps upload`, the new command.

Resolves #10270

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [ ] Other deadline: 
- [x] None: Not urgent, can wait up to 1 week+

## SLA

- Teamwork makes the dream work, so please add a reviewer to your PRs.
- Please give the docs team up to 1 week to review your PR unless you've added an urgent due date to it.
Thanks in advance for your help!

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## EXTRA RESOURCES

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

**Comment by szokeasaurusrex** [comment]
Mobile-team-relevant changes in 676f35ded8151adf78a664aac08f635251e6ac35
Frontend-team-relevant changes in b9e3fd5197220c9a922d591d780ec99c19672c61

File: platform-includes/sourcemaps/legacy-troubleshooting/javascript.mdx
Comment (szokeasaurusrex) [inline platform-includes/sourcemaps/legacy-troubleshooting/javascript.mdx:73]:
```suggestion
sentry-cli sourcemaps upload --url-prefix '~/scripts' .
```

Is the release required here for any specific reason, or was it only present before because it was required with the old API? If we don't need it, let's get rid of it.

This same comment applies to the other changes in this file

File: platform-includes/sourcemaps/legacy-uploading-methods/javascript.mdx
Comment (szokeasaurusrex) [inline platform-includes/sourcemaps/legacy-uploading-methods/javascript.mdx:350]: [Previous comment](https://github.com/getsentry/sentry-docs/pull/10274/files#r1626499445) applies here, as well

**Comment by philipphofmann** [comment]
> Mobile-team-relevant changes in https://github.com/getsentry/sentry-docs/commit/676f35ded8151adf78a664aac08f635251e6ac35

This only affects React-Native, so I leave this to @krystofwoldrich.

**Review by vivianyentran — APPROVED** [review]
content changes lgtm

File: platform-includes/sourcemaps/legacy-troubleshooting/javascript.mdx
Comment (stephanie-anderson) [reply platform-includes/sourcemaps/legacy-troubleshooting/javascript.mdx:73]: @loewenheim maybe do you know the background?

Comment (szokeasaurusrex) [reply platform-includes/sourcemaps/legacy-troubleshooting/javascript.mdx:73]: @loewenheim, so we can likely remove the release now if I understand correctly?

**Comment by szokeasaurusrex** [comment]
@loewenheim, could I get a review from you here?

File: platform-includes/sourcemaps/legacy-troubleshooting/javascript.mdx
Comment (loewenheim) [reply platform-includes/sourcemaps/legacy-troubleshooting/javascript.mdx:73]: Well, it depends on if you're uploading with or without debug ids. Users should ideally use debug ids, but AFAIK it's not mandatory.

**Review by loewenheim — APPROVED** [review]

File: platform-includes/sourcemaps/legacy-troubleshooting/javascript.mdx
Comment (szokeasaurusrex) [reply platform-includes/sourcemaps/legacy-troubleshooting/javascript.mdx:73]: Okay, I will remove `--release` then. I think the CLI will yell at them anyways if they are missing a release when they need it

Comment (loewenheim) [reply platform-includes/sourcemaps/legacy-troubleshooting/javascript.mdx:73]: Sounds good, yeah.

**Pull request merged**
